ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java

```diff
@@ -71,6 +75,20 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  public @Nullable Typeface getTypeface(
+    String fontFamilyName,
+    int style,
+    Context context) {
+    int fontId = context.getResources().getIdentifier(fontFamilyName, "font", context.getPackageName());
```

Comment:
I don't know if the first priority should go to `mFontCache` (which has advantage of it can be changed programmatically) or to the static assets.

Also for static assets, I'd guess both `getIdentifier` and `Typeface.create` call can be slow, that's why there are caches in the first place. If you have 100s of textviews, you probably don't want to redo these computations that never change.

ResourcesCompat is documented as a blocking call. I don't know if that means it blocks for network or something else, but keep that in mind too. I don't have too much knowledge about the new font system, this is just looking up to javadoc, but I don't know if this would then work for downloadable fonts.

Reply:
Maybe adding cache would be a good idea, like you said it'll improve performance for 100s texts.